notebooks/rocoto.ipynb

```diff
@@ -0,0 +1,1013 @@
+    "<!--cell 28-->"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "id": "7b9ecc04-9851-4e34-985f-d908285dc8e2",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "<?xml version='1.0' encoding='utf-8'?>\n",
+      "<workflow realtime=\"False\" scheduler=\"slurm\">\n",
+      "  <cycledef>202410290000 202410300000 06:00:00</cycledef>\n",
+      "  <log>logs/test.log</log>\n",
+      "  <task name=\"bacon\">\n",
+      "    <cores>1</cores>\n",
+      "    <walltime>00:00:10</walltime>\n",
+      "    <command>echo Cooking bacon...</command>\n",
+      "    <jobname>bacon</jobname>\n",
+      "  </task>\n",
+      "  <task name=\"eggs\">\n",
+      "    <nodes>1:ppn=4</nodes>\n",
+      "    <walltime>00:00:10</walltime>\n",
+      "    <command>echo Cooking eggs...</command>\n",
+      "    <jobname>eggs</jobname>\n",
+      "    <dependency>\n",
+      "      <datadep>eggs_recipe.txt</datadep>\n",
+      "    </dependency>\n",
+      "  </task>\n",
+      "  <task name=\"serve\">\n",
+      "    <cores>2</cores>\n",
+      "    <walltime>00:00:01</walltime>\n",
+      "    <command>echo Serving breakfast...</command>\n",
+      "    <jobname>serve</jobname>\n",
+      "    <dependency>\n",
+      "      <and>\n",
+      "        <taskdep task=\"bacon\"/>\n",
+      "        <taskdep task=\"eggs\"/>\n",
+      "      </and>\n",
+      "    </dependency>\n",
+      "  </task>\n",
+      "</workflow>\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%bash\n",
+    "cat tmp/tasks-deps-workflow.xml"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f88d22e0-3433-482d-aeb3-44bbb8cf840f",
+   "metadata": {},
+   "source": [
+    "### Metatasks\n",
+    "\n",
+    "Metatasks define one or more tasks that are similar to one another using a substitution of values. Like tasks, metatask block keys use a suffix after an underscore to name a particular metatask. The metatask in the example below will have a `name=breakfast` attribute in its `<metatask>` tag in the XML document. The values to substitute are defined in a `var:` block, and this block contains one or more keys representing the name of a list of values. The values in the list are separated by spaces. The number of tasks defined by a metatask is equal to the number of values in any list in the `var:` block. In the example below, two lists named `food` and `prepare` contain three values each, so three tasks are defined by this metatask. It is necessary that each list defined in a metatask has the same number of values. The values are referenced using the name of the list that contains the values bracketed by pound signs, as seen in the `task_#food#` key and in the following `command` string. \n",
```

Comment:
I'm curious if it's worth mentioning here that a user would need to nest metatasks to get a combinatorial list of variables. Like if you wanted to bake, cool, and store two different items -- cookies and cakes, you'd need to nest the metatasks:

```
metatask_process:
  var:
    process: bake cool store
  metatask_process_food:
    var:
      food: cookies cakes
...
```

Reply:
I think it was worthwhile to add another example to the metatasks section to show and explain this. I added an example after the initial metatasks example.